### DISCUSSION TITLE
Putting a specific secret in a shared configuration breaks mismatched projects

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#26856] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v37.141.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

We put a secret that we need for a subset of our projects in a shared preset. The secret is encrypted, and it's plaintext has the "o" attribute on it.

What I didn't realize, is that this secret "silently disabled" projects that mismatch the organization attribute: 
```
 INFO: Repository has invalid config (repository=mismatched)
       "error": {
         "validationError": "Failed to decrypt field password. Please re-encrypt and try again.",
         "message": "config-validation",
```

"silently disabled" because it does log the occurrence, but doesn't declare defeat by actually failing. At the same time, it doesn't actually scan anything.

It looks like there is no way of putting that secret in a shared preset AND keeping scanning the projects that mismatch the secret and include that shared preset. I tried moving it into a package rule and the repositories section, but in either case the attempt to decrypt the secret supersedes the filtering applicable configuration.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
This is by design, we don't want projects to be scanned which have a secret loading failure. So the only question is whether the failing needs to be more explicit?

**Reply by astellingwerf** [reply]
I feel it should have been clearer that the project was skipped. The pipeline should have failed, and/or an issue should have been created to report the problem to the repo maintainers.

How can retain a single source of configuration, while dealing with this design decision? How can I prevent secret loading failure for those project to which the secret doesn't apply?

**Reply by rarkins** [reply]
Here's what I'd suggest:
- Have a multi-org shared config without secrets
- For each org, have a shared config (optionally with secrets)
- Repos inherit their org's shared config, not the multi-org shared config directly

BTW I just remembered that a secret can be shared amongst orgs now, but you need to list every org separated by commas, e.g. "org1,org2,org3"

**Reply by astellingwerf** [reply]
The `migratePresets` setting can only be set on a global level, @rarkins? I would need to (ask the owners to) manually change the projects our organization that include our preset to switch to one that contains the cross-org settings?

**Reply by rarkins** [reply]
I don't know
